internal/provider/resource_policy.go

```diff
@@ -0,0 +1,428 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/api/policies"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	policyStorageRetainForDaysKey          = "retain_for_days"
+	policyStorageRetainForOverridableKey   = "retain_for_overridable"
+	policyStorageDeleteAfterDaysKey        = "delete_after_days"
+	policyStorageDeleteAfterOverridableKey = "delete_after_overridable"
+)
+
+func resourcePolicy() *schema.Resource {
```

Comment:
With the TF provider we have been creating sub type specific resources, some of the older types (e.g. targets) we don't do this, but that should be refactored at some point.

So this should be come `resourcePolicyStorage`

Reply:
ok, that makes the policy a bit more straight forward, but do we want the attachment to still be policy type agnostic? or allow for other policy types in the future?